dali/test/python/sequences_test_utils.py

```diff
@@ -220,59 +229,56 @@ def expand_arg(input_layout, num_expand, arg_has_frames, input_batch, arg_batch)
     return expanded_batch
 
 
-def expand_arg_input(input_data, input_layout, num_expand, arg_data, arg_has_frames):
-    assert(len(input_data) == len(arg_data))
-    return [expand_arg(input_layout, num_expand, arg_has_frames, input_batch, arg_batch)
-            for input_batch, arg_batch in zip(input_data, arg_data)]
-
+def expand_arg_input(input_data: ArgData, arg_data: ArgData):
+    assert arg_data.desc.expandable_prefix in ["F", ""]
+    assert len(input_data.data) == len(arg_data.data)
+    arg_has_frames = arg_data.desc.expandable_prefix == "F"
+    return [expand_arg(input_data.desc.expandable_prefix, arg_has_frames, input_batch, arg_batch)
+            for input_batch, arg_batch in zip(input_data.data, arg_data.data)]
 
-def _test_seq_input(device, num_iters, expandable_extents, operator_fn, fixed_params, input_params,
-                    input_layout, input_data, rng):
 
+def _test_seq_input(num_iters, operator_fn, fixed_params, input_params, input_data: ArgData, rng):
     @pipeline_def
-    def pipeline(input_data, input_layout, args_data: List[ArgData]):
-        input = fn.external_source(
-            source=dummy_source(input_data), layout=input_layout)
-        if device == "gpu":
-            input = input.gpu()
+    def pipeline(input_data: ArgData, args_data: List[ArgData]):
```

Comment:
They are the same type, maybe those can be concatenated outside of the pipeline and we would have half of the ifs here removed.

Reply:
Yep, it can. Thanks for catching it that. Somewhere along the refactoring I already tried it but there were some issues with specifying the device, but now it was no problem and it simplified the function indeed,